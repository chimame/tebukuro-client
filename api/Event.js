// @flow
import pathToRegexp from 'path-to-regexp'
import Base from './Base'

export default class Event extends Base {
  cancelRegistration = (params: number) => {
    const url = pathToRegexp.compile(this.endpoints.cancelRegistration)(params)
    return this.client.delete(url).then(this.onSuccess, this.onFailure)
  }
}
