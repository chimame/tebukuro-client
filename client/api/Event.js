import RequestUrls from '../constants/RequestUrls'
import ApiClient from '../utils/ApiClient'

export const createEvent = (params) => {
  return ApiClient.post(`${RequestUrls.communityEvents(params.community_id)}`, params)
}

export const getEvents = (communityId) => {
  return ApiClient.get(`${RequestUrls.communityEvents(communityId)}`)
}

export const showEvent = (id) => {
  return ApiClient.get(`${RequestUrls.events}/${id}`)
}

export const editEvent = (params) => {
  return ApiClient.patch(`${RequestUrls.events}/${params.id}`, params)
}

export const deleteEvent = (id) => {
  return ApiClient.delete(`${RequestUrls.events}/${id}`)
}
