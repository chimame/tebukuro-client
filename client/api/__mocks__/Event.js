const path = require('path')
const Event = jest.genMockFromModule('../Event')

let mockResult = Object.create(null)
export const __setMockResult = (normal = true) => {
  mockResult = normal
}

export const showEvent = (id) => mockResult ? Promise.resolve({message: "normal mock"}) : Promise.reject({message: "error mock"})
export const createEvent = require('../Event').createEvent
export const getEvents = require('../Event').getEvents
export const editEvent = require('../Event').editEvent
export const deleteEvent = require('../Event').deleteEvents
