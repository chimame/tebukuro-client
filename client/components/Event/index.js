import React, { Component } from 'react'

export default class Event extends Component {
  render() {
    return (
      <div>
        <p>name: { this.props.event.name }</p>
        <p>eventStartsAt: { this.props.event.eventStartsAt }</p>
        <p>eventEndsAt: { this.props.event.eventEndsAt }</p>
        <p>description: { this.props.event.description }</p>
        <p>address: { this.props.event.address }</p>
      </div>
    )
  }
}

Event.propTypes = {
  event: React.PropTypes.object.isRequired
}
