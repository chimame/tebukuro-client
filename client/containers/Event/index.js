import React, { Component }   from 'react'
import { connect }            from 'react-redux'
import { bindActionCreators } from 'redux'

export class Event extends Component {
  render() {
    return (
      <div>
        event page sample
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Event)
