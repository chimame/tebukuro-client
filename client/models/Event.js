import Record from './Base'
import { Set } from 'immutable'

export const EventRecord = Record({
  id:             0,
  name:           '',
  description:    '',
  communityId:    0,
  eventStartsAt:  '',
  eventEndsAt:    '',
  address:        '',
  tickets:        Set.of()
})

export default class Event extends EventRecord {
  setTickets(list) {
    return this.set('tickets', Set.of(...list))
  }
}
