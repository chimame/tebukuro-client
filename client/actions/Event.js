import { createAction } from 'redux-actions'
import { createEvent as createAPI, showEvent as showAPI } from '../api/Event'
import Actions from '../constants/Actions'
import errorUrls from '../constants/ErrorUrls'
import ActionDispatch from '../utils/ActionDispatch'

export const createEvent = createAction(Actions.Event.createEvent, createAPI)

export const showEvent = (id) => {
  return ActionDispatch.executeApi(showAPI, id, createAction(Actions.Event.createEvent), errorUrls.pageNotFound)
}
