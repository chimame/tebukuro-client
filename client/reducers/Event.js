import { handleActions }    from 'redux-actions'
import Event                from '../models/Event'
import { ticketReducerMap } from './Ticket'

export const eventInitialState = new Event()

export const eventReducerMap = {
  CREATE_EVENT: {
    next: (state, action) => {
      const event = new Event({
        id:             action.payload.id,
        name:           action.payload.name,
        description:    action.payload.description,
        communityId:    action.payload.community_id,
        eventStartsAt:  action.payload.event_starts_at,
        eventEndsAt:    action.payload.event_ends_at,
        address:        action.payload.address
      })
      return event.setTickets(
        action.payload.tickets ? action.payload.tickets.map((item) => {
          return ticketReducerMap.CREATE_TICKET.next(
            null,
            {payload: item}
          )
        }) : []
      )
    },
    throw: (state, action) => {
      return new Event({
        errors: action.payload.messages
      })
    }
  }
}

export default handleActions(eventReducerMap, eventInitialState)
