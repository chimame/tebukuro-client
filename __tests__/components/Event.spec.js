import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import EventModel from '../../client/models/Event'
import Event from '../../client/components/Event'

const model = (params) => {
  return new EventModel(params)
}

describe('Event', () => {
  it('render event component', () => {
    const eventParams = {
      name: 'rubykansai#77',
      eventStartsAt: '2017-03-04 13:00',
      eventEndsAt: '2017-03-04 17:00',
      description: 'This event is active in Umeda.',
      address: 'Hilton Plaza West 12F'
    }

    const component = shallow(
      <Event event={ model(eventParams) } />
    )

    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
