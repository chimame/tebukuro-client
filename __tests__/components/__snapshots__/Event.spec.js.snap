// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event render event component 1`] = `
<div>
  <p>
    name: 
    rubykansai#77
  </p>
  <p>
    eventStartsAt: 
    2017-03-04 13:00
  </p>
  <p>
    eventEndsAt: 
    2017-03-04 17:00
  </p>
  <p>
    description: 
    This event is active in Umeda.
  </p>
  <p>
    address: 
    Hilton Plaza West 12F
  </p>
</div>
`;
